src/OpenLoco/src/GameCommands/Buildings/CreateBuilding.cpp

```diff
@@ -0,0 +1,264 @@
+#include "CreateBuilding.h"
+#include "Economy/Economy.h"
+#include "Localisation/StringIds.h"
+#include "Map/AnimationManager.h"
+#include "Map/BuildingElement.h"
+#include "Map/IndustryElement.h"
+#include "Map/QuarterTile.h"
+#include "Map/RoadElement.h"
+#include "Map/StationElement.h"
+#include "Map/SurfaceElement.h"
+#include "Map/TileClearance.h"
+#include "Map/TileManager.h"
+#include "Map/TrackElement.h"
+#include "Map/TreeElement.h"
+#include "Objects/BuildingObject.h"
+#include "Objects/LandObject.h"
+#include "Objects/ObjectManager.h"
+#include "Objects/ScaffoldingObject.h"
+#include "S5/S5.h"
+#include "ViewportManager.h"
+#include "World/Industry.h"
+#include "World/Station.h"
+#include "World/TownManager.h"
+
+namespace OpenLoco::GameCommands
+{
+    // 0x0042D133
+    static uint32_t createBuilding(const BuildingPlacementArgs& args, uint8_t flags)
```

Comment:
Does this function require hooking? It seems to be called directly from aiThink. Might be used directly from one of the town growth functions as well.

Reply:
No we have implemented all callers. For once it seems things actually go via doCommand.